### DISCUSSION TITLE
Circular package depedencies with gRPC

### DISCUSSION THREAD

**Discussion by yashykt** [microsoft/vcpkg#27620] [Q&A]
Hello,
I'm new to vcpkg and not quite familiar with things yet, so I apologize for the noobness. 

I work on gRPC and I'm trying to add a target (an extension to gRPC) that depends on `logging` from `google-cloud-cpp`. The issue is that `google-cloud-cpp` depends on `grpc`. So, if I add this new target in `grpc`, it seems like I will have a circular dependency between packages like so -

`grpc` ---> `google-cloud-cpp` ---> `grpc`

The CMake structure in gRPC right now has only a single [CMakelists.txt](https://github.com/grpc/grpc/blob/master/CMakeLists.txt) file which from my understanding translates to a single [package](https://github.com/microsoft/vcpkg/blob/master/ports/grpc/vcpkg.json) in vcpkg. To get around this issue, I'm thinking of creating a separate directory for this extension which will have its own CMake file.

Is my understanding correct?

**Comment by JackBoosY** [comment]
It would be great that gRPC has a CMakeLists now!

My questions is:
1. Is this extension target always depends on `google-cloud-cpp`?
2. Is this extension target always a part of gRPC?

We should handle the circular dependency first.
